contrib/repro_test.bzl

```diff
@@ -0,0 +1,243 @@
+            "%{container_diff_tool}": diff_tool_exec.short_path,
+            "%{img1_path}": img_outs.short_path,
+            "%{img2_path}": img_outs_repro.short_path,
+            "%{img_name}": test_img_label.name,
+        },
+        output = ctx.outputs.test_script,
+        is_executable = True,
+    )
+
+    runfiles = ctx.runfiles(files = [img_outs, img_outs_repro, diff_tool_exec])
+
+    return [
+        DefaultInfo(
+            executable = ctx.outputs.test_script,
+            runfiles = runfiles,
+        ),
+    ]
+
+container_repro_test = rule(
+    attrs = dicts.add(_container.image.attrs, {
+        "test_image": attr.label(
+            doc = "A container_image target to test for reproducibility. " +
+                  "Must be a label relative to the root of the project and " +
+                  "cannot be relative to the container_repro_test target.",
+            mandatory = True,
+        ),
+        "src_project_files": attr.label(
+            allow_files = True,
+            mandatory = True,
+            doc = "Files of the source project where the test_image target " +
+                  "lives. The WORKSPACE file must be specified and is " +
+                  "assumed to be at the root of the project.",
+        ),
+        "base": attr.label(
+            allow_files = container_filetype,
+            mandatory = True,
+            doc = "An image target compatible with the `base` attribute of " +
+                  "the container_image rule to build and reproduce the " +
+                  "test_image in.",
+        ),
+        "container_diff_args": attr.string_list(
+            default = [
+                "--type=file",
+                "--type=size",
+                "--type=apt",
+                "--type=pip",
+            ],
+            doc = "List of arguments to pass to the container_diff tool. " +
+                  "Check https://github.com/GoogleContainerTools/container-diff#container-diff " +
+                  "for more info.",
+        ),
+        "_container_diff_tool": attr.label(
+            allow_single_file = True,
+            cfg = "target",
+            default = Label("@container_diff//file"),
+            executable = True,
+        ),
+        "_test_tpl": attr.label(
+            allow_single_file = True,
+            default = Label("//contrib:cmp_images.sh.tpl"),
+            doc = "A template to expand a bash script to run a complete " +
+                  "image comparison test.",
+        ),
+        "_extract_tpl": attr.label(
+            default = Label("@base_images_docker//util:extract.sh.tpl"),
+            allow_single_file = True,
+        ),
+        "_image_id_extractor": attr.label(
+            allow_single_file = True,
+            default = "//contrib:extract_image_id.py",
+        ),
+    }),
+    implementation = _impl,
+    outputs = dicts.add(_container.image.outputs, {
+        "test_script": "%{name}.test",
+        "img_outs": "%{name}_test_img_outs",
```

Comment:
if these are the outputs from the two container executed call them
img1_outs
img2_outs

Reply:
Done